desk/app/bait.hoon

```diff
@@ -88,53 +105,97 @@
     :_  this
     =/  full-line=request-line:server  (parse-request-line:server url.request)
     =/  line
-      ?:  ?=([%lure @ @ *] site.full-line)
+      ?:  ?=([%lure @ *] site.full-line)
         t.site.full-line
       ?:  ?=([@ @ *] site.full-line)
         site.full-line
       !!
     ?+    method.request  (give not-found:gen:server)
-        %'GET'
-      ?:  ?=([%bait %who ~] line)
-        (give (json-response:gen:server s+(scot %p our.bowl)))
-      =/  inviter  (slav %p i.line)
-      =/  token  i.t.line
-      =/  =metadata:reel  (fall (~(get by token-metadata) [inviter token]) *metadata:reel)
-      ?:  ?=([@ @ %metadata ~] line)
-        (give (json-response:gen:server (enjs-metadata metadata)))
-      (give (manx-response:gen:server (landing-page metadata)))
+      %'GET'  (get-request line)
+    ::
         %'POST'
-      =/  inviter  (slav %p i.line)
-      =/  token  i.t.line
       ?~  body.request
+        ~&  "body not found"
         (give not-found:gen:server)
       ?.  =('ship=%7E' (end [3 8] q.u.body.request))
+        ~&  "ship not found in body"
         (give not-found:gen:server)
       =/  joiner  (slav %p (cat 3 '~' (rsh [3 8] q.u.body.request)))
-      :*  :*  %pass  /bite  %agent  [inviter %reel]
-              %poke  %reel-bite  !>([%bite-1 token joiner inviter])
-          ==
-          :*  %pass  /bite  %agent  [our.bowl %reel]
-              %poke  %reel-bite  !>([%bite-1 token joiner inviter])
-          ==
-          (give (manx-response:gen:server (sent-page joiner)))
-      ==
+      =;  [=bite:reel inviter=(unit ship)]
+        ?~  inviter
+          ~&  "inviter not found"
+          (give not-found:gen:server)
+        ^-  (list card)
+        :*  :*  %pass  /bite  %agent  [u.inviter %reel]
+                %poke  %reel-bite  !>(bite)
+            ==
+            :*  %pass  /bite  %agent  [our.bowl %reel]
+                %poke  %reel-bite  !>(bite)
+            ==
+            (give (manx-response:gen:server (sent-page joiner)))
+        ==
+      =/  =(pole knot)  line
+      ?:  ?=([@ @ ~] line)
+        =/  inviter  (slav %p i.line)
+        =/  old-token  i.t.line
+        :_  `inviter
+        [%bite-1 (scot %t old-token) joiner inviter]
+      =/  token
+        ?~  ext.full-line  i.line
+        (crip "{(trip i.line)}.{(trip u.ext.full-line)}")
+      =/  =metadata:reel  (~(gut by token-metadata) token *metadata:reel)
+      ?~  type=(~(get by fields.metadata) 'bite-type')
+        ~|("no bite type for token: {<token>}" !!)
```

Comment:
We know this isn't a "legitimately-typeless" link because they'll always be of the shape that's caught on line 138 above, right?

Reply:
sure, you think I should just remove the check and assertion underneath?

Reply:
No I think this is fine as is, I was just asking for confirmation.